client/verta/verta/_deployment/endpoint.py

```diff
@@ -9,26 +9,24 @@
 
 
 class Endpoint(object):
-    def __init__(self, conn, conf, workspace, id):
-        self._conn = conn
-        self._conf = conf
-
+    def __init__(self, conn, conf, workspace, id, path):
         self.workspace = workspace
         self.id = id
+        self._path = path
 
     def __repr__(self):
         # TODO: print full info
         return "<Endpoint \"{}\">".format(self.path)
 
     @property
     def path(self):
-        raise NotImplementedError
+        return self._path
```

Comment:
This is essentially what our other objects (`Project`, `RegisteredModel`) do with their `_refresh_cache()`. I think it's fine to make the extra backend call to get `self.path`, to keep a consistent design pattern.
```suggestion
        self._get_json_by_id(…)['creator_…']['path']
```

Reply:
replaced with get json by id